compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -474,12 +474,24 @@ private void addToCompiledSymbols(Set<Symbol> compiledSymbols, Scope.ScopeEntry
             } else {
                 compiledSymbol = symbolFactory.getBCompiledSymbol(symbol, symbol.getOriginalName().getValue());
             }
-            if (compiledSymbol == null || compiledSymbols.contains(compiledSymbol)) {
+            if (compiledSymbol == null || compiledSymbols.contains(compiledSymbol) || isFieldSymbol(compiledSymbol,
+                    isChildOfCurrentNode)) {
                 return;
             }
             compiledSymbols.add(compiledSymbol);
         }
-        addToCompiledSymbols(compiledSymbols, scopeEntry.next, cursorPos, name, states);
+        addToCompiledSymbols(compiledSymbols, scopeEntry.next, cursorPos, name, isChildOfCurrentNode, states);
+    }
+
+    private boolean isFieldSymbol(Symbol compiledSymbol, boolean isChildOfCurrentNode) {
+        switch (compiledSymbol.kind()) {
+            case CLASS_FIELD:
+            case OBJECT_FIELD:
+            case RECORD_FIELD:
+                return isChildOfCurrentNode;
```

Comment:
This is a bit weird/misleading. Basically attempting to do 2 things and misleadingly named.

Reply:
Here the basic intention is to identify if the symbol we've encountered is a field symbol, which is also a field of the current node which the cursor is positioned at.

Those two can be broken down into two different checks. 
Also, if the current node's owner symbol can be passed down to this stage, we could check the owner symbols' equivalency checks once it is verified that it is a field symbol. 

@pubudu91, any other suggestions to improve this proposed filtering mechanism?

Reply:
The high level filtering idea isn't an issue. It's how it's implemented that's problematic. It needs to be refactored in a way that it's not misleading for someone reading it.

Reply:
Refactored the filtering implementation.